src/components/SubjectDiscussion/SubjectDiscussion.js

```diff
@@ -1,21 +1,91 @@
-import { Anchor, Box, Text } from 'grommet'
+import { useEffect, useState } from 'react'
+import { Anchor, Box, Spinner, Text } from 'grommet'
 import { Code as CodeIcon } from 'grommet-icons'
 import styled from 'styled-components'
 
+import { ASYNC_STATES } from '@src/config.js'
 import strings from '@src/strings.json'
+import fetchTalkComments from '@src/helpers/talk/fetchTalkComments.js'
+import fetchUsersById from '@src/helpers/talk/fetchUsersById.js'
+import fetchTalkRoles from '@src/helpers/talk/fetchTalkRoles.js'
 
-const StyledLink = styled(Anchor)`
+import Comment from './Comment.js'
+
+const { READY, FETCHING, ERROR } = ASYNC_STATES
+
+const CleanLink = styled(Anchor)`
   text-decoration: none;
 `
 
 export default function SubjectDiscussion ({
   project, subject
 }) {
+  const [ status, setStatus ] = useState(READY)
+  const [ commentsData, setCommentsData ] = useState([])
+  const [ authors, setAuthors ] = useState(undefined)
+  const [ authorRoles, setAuthorRoles ] = useState(undefined)
+
+  useEffect(function onTargetChange_resetThenFetchData () {
+    setStatus(READY)
+    setCommentsData([])
+    setAuthors([])
+    setAuthorRoles([])
+    doFetchData(subject)
+  }, [project, subject])
```

Comment:
You can avoid blocks like this by adding a key to components. A component will reset when its key changes.